### ISSUE TITLE
Sentry Relay: No events are accepted from CodeOcean

### ISSUE THREAD

**Issue by MrSerth** [openHPI/poseidon#370]
I finished my part for Sentry on CodeOcean and was eager to see it in action. However, the relay doesn't seem to accept any events from CodeOcean -- neither from Ruby nor from JavaScript. Instead, an error is shown in the relay log:

```
public key mismatch on state <public key>
```

I've tried multiple things, but wasn't able to resolve this issue. According to my analysis, everything is correct: The key, the project ID, the slug, ...

The problem is easily reproducible on our staging environment.

**Comment by mpass99** [comment]
You pointed out that the error is thrown only in [line 322](https://github.com/getsentry/relay/blob/075170c9f8c4dbe7ff66779f51f2c2ce2687b8e1/relay-server/src/actors/project.rs#L322).

``` rust
        if !self.is_matching_key(meta.public_key()) {
            relay_log::error!("public key mismatch on state {}", meta.public_key());
```

The function `self.is_matching_key` compares the keys as following

``` rust
if let Some(key_config) = self.get_public_key_config() {
            // Always validate if we have a key config.
            key_config.public_key == project_key
```
whereas `get_public_key_config` just returns the first stored public key.

```rust
    pub fn get_public_key_config(&self) -> Option<&PublicKeyConfig> {
        self.public_keys.get(0)
    }
```

I was able to confirm that our Sentry relay is always working just with the first key, independently of which key is the first.
I assume this is an error in the Sentry Relay.

Anyway, we could use the situation to split the CodeOcean Frontend and Backend into two projects, as otherwise adversaries would be able to spoof Sentry events making it appear that we have issues with our backend..

**Comment by MrSerth** [comment]
Okay, thanks for investigating. I just created another project on Sentry, following your suggestion to split the backend and frontend service. For the CSP reports, I also used the same endpoint as for the browser project.

This change seems to work fine for now, and I am receiving events through the relay. 

Regardless of our change, I wondered whether we should open an issue upstream as the behavior we observed is really unexpected (and annoying).

**Comment by mpass99** [comment]
Submitted @https://github.com/getsentry/relay/issues/2120

**Comment by MrSerth** [comment]
> The issue is now solved by Sentry. Anyway, we keep the current separation into multiple projects, right?

Yes, I would keep the current setup (with multiple projects) at the moment. However, I am still happy that we reported and they fixed it, so that we _could_ use this setup again in the future.
